lib/_stream_writable.js

```diff
@@ -294,34 +294,32 @@ function validChunk(stream, state, chunk, cb) {
 
 Writable.prototype.write = function(chunk, encoding, cb) {
   const state = this._writableState;
-  var ret = false;
-  const isBuf = !state.objectMode && Stream._isUint8Array(chunk);
```

Comment:
@addaleax: I don't quite understand the thing with _isUint8Array.

Shouldn't this:
```js
const isBuf = !state.objectMode && Stream._isUint8Array(chunk);
```
be:
```js
const isBuf = !state.objectMode && (chunk instanceof Buffer || Stream._isUint8Array(chunk));
```
Note the chunk instanceof Buffer.

Currently only Uint8Array will get encoding buffer while Buffer will not.

Reply:
`ObjectSetPrototypeOf(Buffer, Uint8Array);`